### ISSUE TITLE
<Modal> removes and re-adds components to the DOM when closing

### ISSUE THREAD

**Issue by buckett** [instructure/instructure-ui#96]
We had a modal component that contained a component that loaded data using `fetch()` in `componentDidMount()` and were noticing that when the modal was closed it would re-fetch the data. Adding some logging to the constructor of component doing the fetch showed it being recreated when the modal was closed. To work around this issue we just set the transition on the modal to be `null` as then we don't see this.

I think the problem is:

1. The props on `<Modal>` get updated to have `open=false`, this causes the `componentDidUpdate()` on `<Modal>` to get called which calls `setState({})` but this doesn't actually set the state at this point (so it's not yet transitioning).
2. The component gets re-rendered and it's closed and not transitioning so the contents of the modal are removed from the DOM.
3. The `setState()` get committed and this causes the modal to re-render, now it is transitioning so it adds the elements back to the DOM while the transitioning is happening.
4. The transition is complete and it's no longer transitioning so the elements are removed from the DOM.

This is just my theory at the moment as I haven't double checked.

**Comment by buckett** [comment]
We are also seeing a similar issue with <Tabs> that use <Transition> under the hood.

**Comment by buckett** [comment]
Sorry, is this fixed? we're still seeing this issue with the latest release.

**Comment by HerrTopi** [comment]
No, it was closed by mistake. We did a cleanup in the backlog and accidentally closed this as well. Thanks for catching it.

**Comment by matyasf** [comment]
Hi

We've fixed this issue in the latest release (7.3.5)

**Comment by buckett** [comment]
Thanks, is 397c2b87a41a565d5a16afb56529a2b8da1fc8d7 the fix?

**Comment by matyasf** [comment]
Yes, and the next commit on the `Master` branch
